File: src/main/java/com/example/leaderboard/domain/user/controller/UserController.java
package com.example.leaderboard.domain.user.controller;

import com.example.leaderboard.common.response.ApiResponse;
import com.example.leaderboard.domain.user.dto.UserCreateRequest;
import com.example.leaderboard.domain.user.dto.UserResponse;
import com.example.leaderboard.domain.user.service.UserService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/v1/users")
@RequiredArgsConstructor
public class UserController {

    private final UserService userService;

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public ApiResponse<UserResponse> createUser(@Valid @RequestBody UserCreateRequest request) {
        return ApiResponse.ok(userService.createUser(request));
    }

    @GetMapping("/{id}")
    public ApiResponse<UserResponse> getUser(@PathVariable Long id) {
        return ApiResponse.ok(userService.getUser(id));
    }

    @GetMapping
    public ApiResponse<List<UserResponse>> getUsers() {
        return ApiResponse.ok(userService.getUsers());
    }
}

File: src/main/kotlin/com/example/leaderboard/domain/user/controller/UserController.kt
package com.example.leaderboard.domain.user.controller

import com.example.leaderboard.common.response.ApiResponse
import com.example.leaderboard.domain.user.dto.UserCreateRequest
import com.example.leaderboard.domain.user.dto.UserResponse
import com.example.leaderboard.domain.user.service.UserService
import jakarta.validation.Valid
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/v1/users")
class UserController(private val userService: UserService) {

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    fun createUser(@Valid @RequestBody request: UserCreateRequest): ApiResponse<UserResponse> =
        ApiResponse.ok(userService.createUser(request))

    @GetMapping("/{id}")
    fun getUser(@PathVariable id: Long): ApiResponse<UserResponse> =
        ApiResponse.ok(userService.getUser(id))

    @GetMapping
    fun getUsers(): ApiResponse<List<UserResponse>> =
        ApiResponse.ok(userService.getUsers())
}
